add a compat test case which verifies that shallow clones can be deep…

stspdotname opened the pull request:
…ened

This test shows that deepening shallow clones works across all Git protocol versions. The trick to make this work is to pass depth=DEPTH_INFINITE

Related to issue #1399

jelmer commented:
Please rebase - the ruff issues are fixed in master.

stspdotname commented:
These commits are already based on https://github.com/jelmer/dulwich/commit/4de310b6676856434391ab57b55e0e4489967abe
which seems to be the latest at present?

jelmer commented:
> These commits are already based on [4de310b](https://github.com/jelmer/dulwich/commit/4de310b6676856434391ab57b55e0e4489967abe) which seems to be the latest at present?

Whoops, the PR actually got stuck because of a spurious test failure. Current HEAD should be good.

stspdotname committed:
add a compat test case which verifies that shallow clones can be deepened

This test shows that deepening shallow clones works across all Git protocol
versions. The trick to make this work is to pass depth=DEPTH_INFINITE

Related to issue #1399

jelmer commented:
Changes LGTM, just needs a "ruff format" run?

jelmer commented:
FWIW it looks like these test failures are repeatable, and appear in the newly added tests. They might be related to general brittleness.

jelmer committed:
Recreate client, needed for TCP clients
